Library/ViewModels/DiscoveryViewModel.swift

```diff
@@ -204,7 +231,10 @@ public final class DiscoveryViewModel: DiscoveryViewModelType, DiscoveryViewMode
   }
 
   fileprivate let willTransitionToPageProperty = MutableProperty<Int>(0)
+  fileprivate let previouslyTransitionedToPageProperty =
+    MutableProperty<Int>(1) // Store last page before new page transition
   public func willTransition(toPage nextPage: Int) {
+    self.previouslyTransitionedToPageProperty.value = self.willTransitionToPageProperty.value
```

Comment:
Same idea as above. I think you you should try to create a new input/output to handle the logic you are introducing.
https://github.com/kickstarter/native-docs/blob/master/inputs-outputs.md

Reply:
Made some changes to this.